### PR TITLE
PointerDeviceKind forwards-compatibility in flutter_test

### PR DESCRIPTION
**Pull request by moffatman**
This use of PointerDeviceKind was missed in https://github.com/flutter/flutter/pull/97350

Needed to reland https://github.com/flutter/engine/pull/28571
Original issue is https://github.com/flutter/flutter/issues/23604

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by dkwingsmt — APPROVED** [review]
LGTM, need a second review

**Review by Hixie — APPROVED** [review]
LGTM

test-exempt: temporary avoidance of analysis errors to allow code to land

**Pull request merged**
